### PR TITLE
Salt changes to enable using non-default Kubelet port

### PR DESCRIPTION
**Pull request by gmarek**
cc @liggitt

### CODE REVIEW

**Comment by lavalamp** [comment]
@roberthbailey, can you find someone to review? (Or maybe @thockin can?)

File: cluster/gce/trusty/node.yaml [hunk 1]
```diff
@@ -262,7 +262,7 @@ script
 			echo "Docker daemon failed!"
 			pkill docker
 		fi
-		if ! curl --insecure -m ${max_seconds} -f -s https://127.0.0.1:10250/healthz > /dev/null; then
+		if ! curl --insecure -m ${max_seconds} -f -s https://127.0.0.1:${KUBELET_PORT}/healthz > /dev/null; then
```
Comment (roberthbailey) [inline cluster/gce/trusty/node.yaml]: This doesn't handle proper defaulting if KUBELET_PORT isn't explicitly specified.

**Comment by roberthbailey** [comment]
The salt changes look ok, but I'm worried that the change to the trusty configs won't work on GKE.

**Comment by andyzheng0831** [comment]
@roberthbailey, you are right this will break the health monitoring in trusty. Although KUBELET_PORT is exported as an entry of kube-env in debian/helper.sh, but it is not sourced in trusty/node.yaml.

@gmarek, please add the following two lines above where you changed trusty/node.yaml
# Configuration files are located at /etc/saltbase.
. /etc/kube-env

**Comment by thockin** [comment]
Sadly, this will cause a (minor) problem in the Service controller if people use it - we have 10250 blacklisted for Service IPs with load-balancers (because GCE's firewall is broken). It's a minor point, but it exists...

File: cluster/gce/trusty/node.yaml (hunk 1 above)
Comment (gmarek) [inline cluster/gce/trusty/node.yaml]: Done.

**Comment by gmarek** [comment]
@andyzheng0831 - Done.

@thockin - IIUC this is the problem with running Kubelet on non-default port despite of this change, right? Currently it's also possible to change the kubelet port (there's a flag for it), but it'll break API server proxy. Because of that I'd argue that this change makes it strictly better.

**Comment by liggitt** [comment]
> Because of that I'd argue that this change makes it strictly better.

Agree. If the port is required to remain at 10250 for specific environments, this allows that, but it plumbs support for alternate ports for other environments, right?

**Comment by andyzheng0831** [comment]
@gmarek, I pasted a wrong line "# Configuration files are located at /etc/saltbase." in my previous suggestion. That line is confusing and not needed. Please remove it, and just keep ". /etc/kube-env" is adequate. Sorry about it.

**Comment by gmarek** [comment]
Done.

**Comment by gmarek** [comment]
@liggitt - right.

**Comment by andyzheng0831** [comment]
LGTM

**Comment by roberthbailey** [comment]
LGTM.

**Comment by thockin** [comment]
LGTM

**Comment by wojtek-t** [comment]
@gmarek - please rebase

**Comment by gmarek** [comment]
Reverting because it probably broke GKE build. @roberthbailey - can you fix this (or delegate) on the GKE side?

**Comment by roberthbailey** [comment]
@jlowdermilk did you notice the broken GKE e2e tests?

**Comment by jlowdermilk** [comment]
I did not. I wasn't watching them very closesly at the time, as the entire framework was broken for a couple days.

**Comment by gmarek** [comment]
gke-ci runs 9683 to 9689.

**Pull request merged**